scheduler/src/main/java/gov/nasa/jpl/aerie/scheduler/model/ActivityInstance.java

```diff
@@ -362,15 +362,15 @@ public Map<String, VariableArgumentComputer> getVariableArguments() {
    * @param arguments a name/value map of arguments
    */
   public void setArguments(Map<String, SerializedValue> arguments) {
-    this.arguments = arguments;
+    this.arguments.putAll(arguments);
```

Comment:
Is setArguments expected to remove all existing arguments, or is it an overlay?

E.g. if this.arguments has { "a": 1 }, and we call setArguments({ "b": 2 }), do we expect the arguments to have both keys, or just "b"? The current implementation will have both keys.

Reply:
The two options I see here are either clearing the map before calling `putAll` (if it's important that the reference remain the same)
```suggestion
    this.arguments.clear();
    this.arguments.putAll(arguments);
```

Or, if it's okay to create a new reference, we could simply create a new map:
```suggestion
    this.arguments = new HashMap<>(arguments);
```

Reply:
You are absolutely right. I'll make the change.